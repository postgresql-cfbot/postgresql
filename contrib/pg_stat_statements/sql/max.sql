SHOW pg_stat_statements.max;

SET pg_stat_statements.track = 'all';

DO $$
BEGIN
  FOR i IN 1..101 LOOP
    EXECUTE format('create table t%s (a int)', lpad(i::text, 3, '0'));
  END LOOP;
END
$$;

SELECT pg_stat_statements_reset() IS NOT NULL AS t;

DO $$
BEGIN
  FOR i IN 1..98 LOOP
    EXECUTE format('select * from t%s', lpad(i::text, 3, '0'));
  END LOOP;
END
$$;

SELECT count(*) <= 100 AND count(*) > 0 FROM pg_stat_statements;
SELECT query FROM pg_stat_statements WHERE query LIKE '%t001%' OR query LIKE '%t098%' ORDER BY query;

DO $$
BEGIN
  FOR i IN 2..98 LOOP
    EXECUTE format('select * from t%s', lpad(i::text, 3, '0'));
  END LOOP;
END
$$;

DO $$
BEGIN
  FOR i IN 99..100 LOOP
    EXECUTE format('select * from t%s', lpad(i::text, 3, '0'));
  END LOOP;
END
$$;

SELECT count(*) <= 100 AND count(*) > 0 FROM pg_stat_statements;
-- record for t001 has been kicked out
SELECT query FROM pg_stat_statements WHERE query LIKE '%t001%' ORDER BY query;
