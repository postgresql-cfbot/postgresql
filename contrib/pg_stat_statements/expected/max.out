SHOW pg_stat_statements.max;
 pg_stat_statements.max 
------------------------
 100
(1 row)

SET pg_stat_statements.track = 'all';
DO $$
BEGIN
  FOR i IN 1..101 LOOP
    EXECUTE format('create table t%s (a int)', lpad(i::text, 3, '0'));
  END LOOP;
END
$$;
SELECT pg_stat_statements_reset() IS NOT NULL AS t;
 t 
---
 t
(1 row)

DO $$
BEGIN
  FOR i IN 1..98 LOOP
    EXECUTE format('select * from t%s', lpad(i::text, 3, '0'));
  END LOOP;
END
$$;
SELECT count(*) <= 100 AND count(*) > 0 FROM pg_stat_statements;
 ?column? 
----------
 t
(1 row)

SELECT query FROM pg_stat_statements WHERE query LIKE '%t001%' OR query LIKE '%t098%' ORDER BY query;
       query        
--------------------
 select * from t001
 select * from t098
(2 rows)

DO $$
BEGIN
  FOR i IN 2..98 LOOP
    EXECUTE format('select * from t%s', lpad(i::text, 3, '0'));
  END LOOP;
END
$$;
DO $$
BEGIN
  FOR i IN 99..100 LOOP
    EXECUTE format('select * from t%s', lpad(i::text, 3, '0'));
  END LOOP;
END
$$;
SELECT count(*) <= 100 AND count(*) > 0 FROM pg_stat_statements;
 ?column? 
----------
 t
(1 row)

-- record for t001 has been kicked out
SELECT query FROM pg_stat_statements WHERE query LIKE '%t001%' ORDER BY query;
 query 
-------
(0 rows)

